loginCore/src/test/java/android/text/TextUtils.java

```diff
@@ -0,0 +1,14 @@
+package android.text;
+
+public class TextUtils {
+    public static boolean isDigitsOnly(CharSequence str) {
```

Comment:
Copying java code is a no no. This can be done much more elegantly in Kotlin

Reply:
Fixed by using a new extension function `String.isLoginCode()` in the extensions file. It is essentially `.isDigitsOnly()` by is a Kotlin version that can be used in unit tests without needing to be mocked.